src/WebJobs.Script/Diagnostics/Extensions/LoggerExtension.cs

```diff
@@ -121,15 +121,15 @@ internal static class LoggerExtension
                 new EventId(325, nameof(ScriptStartUpLoadedExtension)),
                 "Loaded extension '{startupExtensionName}' ({startupExtensionVersion})");
 
-        private static readonly Action<ILogger, Exception> _functionMetadataProviderReadingMetadata =
```

Comment:
It's hard without tests to see what the results of these changes are. I see there's a FunctionMetadataProviderTests suite that is exercising worker/host indexing scenarios, but perhaps those tests aren't validating these logs.

I think we should update those tests so they're validating these logs, and or also add a test in there that repros the issue described in https://github.com/Azure/azure-functions-host/issues/9420 and validates/demonstrates the fix.

Reply:
Updated some existing tests to also validate loggings